src/MonitorProfiler/Communication/IpcCommClient.cpp

```diff
@@ -59,15 +103,53 @@ HRESULT IpcCommClient::Send(const IpcMessage& message)
         return E_UNEXPECTED;
     }
 
-    char buffer[sizeof(MessageType) + sizeof(int)];
-    *reinterpret_cast<MessageType*>(buffer) = message.MessageType;
-    *reinterpret_cast<int*>(&buffer[sizeof(MessageType)]) = message.Parameters;
+    if (message.Payload.size() > UINT32_MAX)
```

Comment:
As with my previous comment, we should establish some reasonable message size and not allow anything larger.